web/static/dlg_start_xfer.js

```diff
@@ -81,206 +82,193 @@ export function show( a_mode, a_ids, a_cb ) {
             selectMode: 3,
-            icon:false,
-            select:function( ev, data ){
+            icon: false,
+            select: function (ev, data) {
                 var sel = rec_tree.getSelectedNodes();
-                if ( sel.length ){
-                    if ( !selection_ok ){
+                if (sel.length) {
+                    if (!selection_ok) {
                         selection_ok = true;
                         updateGoBtn();
                     }
-                }else{
-                    if ( selection_ok ){
+                } else {
+                    if (selection_ok) {
                         selection_ok = false;
                         updateGoBtn();
                     }
                 }
             }
         });
 
-        $("#records",frame).show();
-        rec_tree = $.ui.fancytree.getTree($("#records",frame));
-    }else{
+        $("#records", frame).show();
+        rec_tree = $.ui.fancytree.getTree($("#records", frame));
+    } else {
         item = a_ids[0];
-        console.log("item:",item);
+        console.log("item:", item);
 
-        var html = item.id + "&nbsp&nbsp" + util.sizeToString( item.size ) + "&nbsp&nbsp" + util.escapeHTML( item.title );
-        $("#title",frame).html( html );
-        if ( a_mode == model.TT_DATA_PUT && item.source ){
-            console.log("source:",item.source);
-            path_in.val( item.source );
+        var html = item.id + "&nbsp&nbsp" + util.sizeToString(item.size) + "&nbsp&nbsp" + util.escapeHTML(item.title);
+        $("#title", frame).html(html);
+        if (a_mode == model.TT_DATA_PUT && item.source) {
+            console.log("source:", item.source);
+            path_in.val(item.source);
         }
     }
 
     util.inputTheme(path_in);
-    util.inputTheme($("#ext",frame));
+    util.inputTheme($("#ext", frame));
 
-    matches.on('selectmenuchange', function( ev, ui ) {
-        if ( ep_list && ui.item ){
-            var ep = ep_list[ui.item.index-1].id;
-            api.epView( ep, function( ok, data ){
-                if ( ok && !data.code ){
+    matches.on('change', function (ev) {
+        if (ep_list) {
+            var ep = ep_list[$(this).prop('selectedIndex') - 1].id;
+            api.epView(ep, function (ok, data) {
+                if (ok && !data.code) {
                     cur_ep = data;
                     cur_ep.name = cur_ep.canonical_name || cur_ep.id;
-                    path_in.val( cur_ep.name + (cur_ep.default_directory?cur_ep.default_directory:"/"));
-                    updateEndpointOptions( cur_ep );
-                }else{
-                    dialogs.dlgAlert("Globus Error", data );
+                    path_in.val(cur_ep.name + (cur_ep.default_directory ? cur_ep.default_directory : "/"));
+                    updateEndpointOptions(cur_ep);
+                } else {
+                    dialogs.dlgAlert("Globus Error", data);
                 }
             });
         }
     });
```

Comment:
Put a comment above function describing what it is doing. If possible give an example of what the input fields should be and explain the expected behavior.